design/baremetal-operator/firmware-interface.md

```diff
@@ -0,0 +1,221 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# firmware-interface
+
+## Status
+
+implementable
+
+## Summary
+
+This design document describes a Metal3 API for using the Firmware Interface
+feature added to Ironic, this will allow updates of firmware of bmc and bios
+in BMH that are compatible with Redfish.
+
+## Motivation
+
+When provisioning new machines to clusters, it may be necessary to upgrade
+or downgrade the servers BIOS and firmware to specific versions to ensure
+the server is configured as it is in their validated pattern.
+
+Redfish has a common API that allow users to execute a firmware update, the
+[proposal](https://review.opendev.org/c/openstack/ironic-specs/+/878505) was
```

Comment:
nit: link to upstream docs once we have them

Reply:
should be https://specs.openstack.org/openstack/ironic-specs/priorities/2023-2-workitems.html#firmware-updates

Reply:
I was thinking about adding the spec that got merged, but maybe is ok to have the workitems url =)